## Installation of software in a custom directory

In
```
cd /fs/ess/scratch/PAS2303/Jonathan/Annotation/Software
```

Installing **BBTools**: a suite of fast, multithreaded bioinformatics tools designed for analysis of DNA and RNA sequence data https://jgi.doe.gov/data-and-tools/bbtools/
```
wget https://sourceforge.net/projects/bbmap/files/BBMap_39.00.tar.gz
tar -xvzf BBMap_39.00.tar.gz
rm BBMap_39.00.tar.gz
# testing BBTools:
bash ./bbmap/bbmerge.sh
```

Installing **FastQC v0.11.9** https://www.bioinformatics.babraham.ac.uk/projects/fastqc/
```
apptainer pull fastqc.sif docker://quay.io/biocontainers/fastqc:0.11.9--hdfd78af_1
# Does FastQC work?
apptainer exec fastqc.sif fastqc --help
```

Installing **MultiQC v1.13** https://multiqc.info 
```
apptainer pull MultiQC.sif docker://quay.io/biocontainers/multiqc:1.13--pyhdfd78af_0
# Does MultiQC work?
apptainer exec MultiQC.sif multiqc --help

```

Installing **Funannotate 1.8.13**: Fuannotate: pipeline for annotating fungal genomes https://github.com/nextgenusfs/funannotate
```
#apptainer pull Funannotate.sif docker://nextgenusfs/funannotate
apptainer pull Funannotate.sif docker://reslp/funannotate:latest 
# Testing Funannotate
apptainer exec Funannotate.sif funannotate
```

Installing **GeneMark-ES/ET**: (you will need to generate your own license, Jonathan will show you how to do that) Alexandre Lomsadze et al Gene identification in novel eukaryotic genomes by self-training algorithm http://nar.oxfordjournals.org/content/33/20/6494.abstract  http://topaz.gatech.edu/GeneMark/license_download.cgi 
```
wget http://topaz.gatech.edu/GeneMark/tmp/GMtool_goMiJ/gmes_linux_64_4.tar.gz
tar -xvzf gmes_linux_64_4.tar.gz
rm gmes_linux_64_4.tar.gz
cd gmes_linux_64_4
wget http://topaz.gatech.edu/GeneMark/tmp/GMtool_goMiJ/gm_key_64.gz
gunzip gm_key_64.gz
cp gm_key_64 ~/.gm_key
``` 

Installing **Agat v0.9.2**: Another Gtf/Gff Analysis Toolkit https://github.com/NBISweden/AGAT
```
apptainer pull AGAT.sif docker://quay.io/biocontainers/agat:0.9.2--pl5321hdfd78af_1
```

Installing **BUSCO v5.4.3**: Another Gtf/Gff Analysis Toolkit https://github.com/NBISweden/AGAT
```
apptainer pull BUSCO.sif docker://quay.io/biocontainers/busco:5.4.3--pyhdfd78af_0
```




Installing **eggnog-mapper v2.1.9**: fast functional annotation of novel sequences https://github.com/eggnogdb/eggnog-mapper
```
wget  https://github.com/eggnogdb/eggnog-mapper/archive/refs/tags/2.1.9.tar.gz
tar -pxvzf 2.1.9.tar.gz
rm 2.1.9.tar.gz
cd eggnog-mapper-2.1.9
module load python/3.9-2022.05
python setup.py install --prefix /fs/ess/scratch/PAS2303/Jonathan/Annotation/Software/eggnog-mapper-2.1.9/
export PATH=/fs/ess/scratch/PAS2303/Jonathan/Annotation/Software/eggnog-mapper-2.1.9:/fs/ess/scratch/PAS2303/Jonathan/Annotation/Software/eggnog-mapper-2.1.9/bin:"$PATH"
export EGGNOG_DATA_DIR=/fs/ess/scratch/PAS2303/Jonathan/Annotation/Software/eggnog-mapper-2.1.9/data
python download_eggnog_data.py -P -M -H -d Fungi
create_dbs.py -m diamond --dbname fungi --taxa Fungi
#python download_eggnog_data.py -P -M -H -d Viridiplantae
#create_dbs.py -m diamond --dbname viridiplantae --taxa Viridiplantae
# Checking if it works:
emapper.py -h
```

Installing **antiSMASH v6.1.1**: fast functional annotation of novel sequences https://github.com/eggnogdb/eggnog-mapper
```
apptainer pull antiSMASH.sif docker://quay.io/biocontainers/antismash:6.1.1--pyhdfd78af_0
```


Installing **PASA**:  Program to Assemble Spliced Alignments https://github.com/PASApipeline/PASApipeline/
```
module load gnu/11.2.0
wget https://github.com/PASApipeline/PASApipeline/releases/download/pasa-v2.5.2/PASApipeline-v2.5.2.FULL.tar.gz
tar -xvzf PASApipeline-v2.5.2.FULL.tar.gz
rm PASApipeline-v2.5.2.FULL.tar.gz
cd PASApipeline-v2.5.2
make
# testing PASA
cd ..
/fs/ess/scratch/PAS2303/Jonathan/Annotation/Software/PASApipeline-v2.5.2/bin/pasa
```

Installing **Trinity 2.14**: Trinity: reconstructing a full-length transcriptome without a genome from RNA-Seq data https://doi.org/10.1038/nbt.1883 for the Protocol: https://doi.org/10.1038/nprot.2013.084
```
wget https://github.com/trinityrnaseq/trinityrnaseq/releases/download/Trinity-v2.14.0/trinityrnaseq-v2.14.0.FULL.tar.gz
tar -xvzf trinityrnaseq-v2.14.0.FULL.tar.gz
rm trinityrnaseq-v2.14.0.FULL.tar.gz
cd trinityrnaseq-v2.14.0
module load  cmake/3.17.2
make
# testing Trinity
cd ..
./trinityrnaseq-v2.14.0/Trinity
```
